scripts/setup.py

```diff
@@ -258,6 +258,7 @@ def setup(runner_type: Type[Setup]) -> None:
     parser.add_argument("--development", action="store_true")
     parser.add_argument("--release", action="store_true")
     parser.add_argument("--build-type", type=BuildType)
+    parser.add_argument("--no-tests", action="store_true")
```

Comment:
Why do we need this?

Reply:
These tests are for validating the compiled Makefile and stuff.
It take a lot of time and prolong the action further. But if you want, we can run the tests!

This flag builds Makefile without building and running the tests saving us time.